File: 350.intersection-of-two-arrays-ii.java
import java.util.ArrayList;

/*
 * @lc app=leetcode id=350 lang=java
 *
 * [350] Intersection of Two Arrays II
 */

// @lc code=start
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        int[] cnt = new int[1001];
        for (int num : nums1) {
            cnt[num]++;
        }
        ArrayList<Integer> list = new ArrayList<>();
        for (int num : nums2) {
            if (cnt[num] > 0) {
                list.add(num);
                cnt[num]--;
            }
        }
        int[] res = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            res[i] = list.get(i);
        }
        return res;
    }
}
// @lc code=end

File: 350.intersection-of-two-arrays-ii.py
#
# @lc app=leetcode id=350 lang=python3
#
# [350] Intersection of Two Arrays II
#

# @lc code=start

class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        count1=Counter(nums1)
        count2=Counter(nums2)
        res=[]
        for num in count1:
            if num in count2:
                length=min(count1[num], count2[num])
                for i in range(length):
                    res.append(num)

        return res

# @lc code=end

